File: PHP/Pc.php
<?php

/* Saya Listia Ningrum NIM 2008084 mengerjakan Latihan 3 dalam mata
    kuliah DPBO untuk keberkahanNya maka saya tidak akan melakukan kecurangan 
	seperti yang telah dispesifikasikan. Aamiin */

include "Processor.php";
include "Disk.php";
include "Ram.php";

class Pc
{
	// deklarasi atribut
	private $pObj;
    private $dObj;	
    private $rObj;	
	private $totalPrice;
	
	// constructor
	public function __construct(){
		$this -> pObj = new Processor();
		$this -> dObj = new Disk();
		$this -> rObj = new Ram();
	}
	
	// method set & get processor
	public function setProcessor($pObj){
		$this -> pObj = $pObj;
	}
	public function getProcessor(){
		return $this -> pObj;
	}

	// method set & get disk
	public function setDisk($dObj){
		$this -> dObj = $dObj;
	}
	public function getDisk(){
		return $this -> dObj;
	}

	// method set & get ram
	public function setRam($rObj){
		$this -> rObj = $rObj;
	}
	public function getRam(){
		return $this -> rObj;
	}

	// method set & get total price
	public function setTotalPrice($pObj, $dObj, $rObj){
		$this -> totalPrice = $pObj -> getPrice() + $dObj -> getPrice() + $rObj -> getPrice();
    }
	public function getTotalPrice(){
		return $this -> totalPrice;
	}
}

?>

File: Java/Pc.java
/* Saya Listia Ningrum NIM 2008084 mengerjakan Latihan 3 dalam mata
	kuliah DPBO untuk keberkahanNya maka saya tidak akan melakukan kecurangan 
	seperti yang telah dispesifikasikan. Aamiin */

class Pc
{
    // deklarasi atribut
    private Processor pObj;
    private Disk dObj;
    private Ram rObj;
    private int totalPrice;

    // constructor
    public Pc(){   
        this.pObj = new Processor();
		this.dObj = new Disk();
		this.rObj = new Ram();
    }
 
    // method set & get processor
	public void setProcessor(Processor pObj){
		this.pObj = pObj;
	}
	public Processor getProcessor(){
		return pObj;
	}

	// method set & get disk
	public void setDisk(Disk dObj){
		this.dObj = dObj;
	}
	public Disk getDisk(){
		return dObj;
	}

	// method set & get ram
	public void setRam(Ram rObj){
		this.rObj = rObj;
	}
	public Ram getRam(){
		return rObj;
	}

	// method set & get total price
	public void setTotalPrice(Processor pObj, Disk dObj, Ram rObj){
		this.totalPrice = pObj.getPrice() + dObj.getPrice() + rObj.getPrice();
    }
	public int getTotalPrice(){
		return totalPrice;
	}
}